lib/eventmachine.rb

```diff
@@ -184,7 +184,15 @@ def self.run blk=nil, tail=nil, &block
           add_timer(0) { signal_loopbreak }
         end
         @reactor_thread = Thread.current
-        run_machine
+
+        # Rubinius needs to come back into "Ruby space" for GC to work,
+        # so we'll crank the machine here.
+        if defined?(RUBY_ENGINE) && RUBY_ENGINE == "rbx"
+          while run_machine_once; end
```

Comment:
Just the tiniest of style notes, I like to use `true while run_machine_once` for this kind of loop.